examples/property_prediction/pretrain_gnns/chem/pretrain_supervised.py

```diff
@@ -0,0 +1,144 @@
+            for step, (bg, labels) in enumerate(tq_train):
+                node_feats = [
+                    bg.ndata.pop('atomic_number'),
+                    bg.ndata.pop('chirality_type')
+                ]
+                edge_feats = [
+                    bg.edata.pop('bond_type'),
+                    bg.edata.pop('bond_direction_type')
+                ]
+                bg = bg.to(device)
+                node_feats = [n.to(device) for n in node_feats]
+                edge_feats = [e.to(device) for e in edge_feats]
+
+                logits = model(bg, node_feats, edge_feats)
+
+                is_valid = labels ** 2 > 0
+                is_valid = is_valid.to(device)
+                labels = labels.type_as(logits)
+                loss = criterion(logits, (labels + 1) / 2)
+                loss = torch.where(is_valid, loss, torch.zeros(loss.shape).to(loss.device).to(loss.dtype))
+                loss = torch.sum(loss) / torch.sum(is_valid)
+
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+                tq_train.set_postfix_str(
+                    "Epoch: {} Step: {} Loss: {:.4f}".format(epoch,
+                                                             step + 1,
+                                                             loss.item()), refresh=False)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='pretrain_supervised')
+    parser.add_argument('--device', type=int, default=0,
+                        help='which gpu to use if any (default: 0)')
+    parser.add_argument('--batch_size', type=int, default=32,
+                        help='input batch size for training (default: 32)')
+    parser.add_argument('--epochs', type=int, default=100,
+                        help='number of epochs to train (default: 100)')
+    parser.add_argument('--lr', type=float, default=0.001,
+                        help='learning rate (default: 0.001)')
+    parser.add_argument('--decay', type=float, default=0,
+                        help='weight decay (default: 0)')
+    parser.add_argument('--num_layer', type=int, default=5,
+                        help='number of GNN message passing layers (default: 5).')
+    parser.add_argument('--emb_dim', type=int, default=300,
+                        help='embedding dimensions (default: 300)')
+    parser.add_argument('--dropout_ratio', type=float, default=0.2,
+                        help='dropout ratio (default: 0.2)')
+    parser.add_argument('--graph_pooling', type=str, default="mean",
+                        help='graph level pooling (sum, mean, max, set2set, attention)')
+    parser.add_argument('--JK', type=str, default="last",
+                        help='how the node features across layers are combined. last, sum, max or concat')
+    parser.add_argument('--dataset', type=str, default='chembl_filtered',
+                        help='root directory of dataset. For now, only classification.')
```

Comment:
By "root directory of dataset", do you want to support custom datasets? If so, you should explicitly state the format requirement in README.

Reply:
done